src/inference/mh.jl

```diff
@@ -69,45 +70,26 @@ Perform a generalized (reversible jump) Metropolis-Hastings update based on an i
 
 The `involution` is a callable value that has the following signature:
 
-    (new_trace, bwd_choices::ChoiceMap, weight) = involution(trace, fwd_choices::ChoiceMap, fwd_ret, proposal_args::Tuple; check::Bool=false)
+    (new_trace, bwd_choices::ChoiceMap, weight) = involution(trace::Trace, fwd_trace::Trace; check::Bool=false)
 
-Most users will want to construct `involution` using the [Involution DSL](@ref) with the [`@involution`](@ref) macro, but is also possible to provide a Julia function for `involution`.
+Most users will want to construct `involution` using the [Bijection DSL](@ref) with the [`@bijection`](@ref) macro, but is also possible to provide a Julia function for `involution`.
 
 The generative function `proposal` is executed on arguments `(trace, proposal_args...)`, producing a choice map `fwd_choices` and return value `fwd_ret`.
 For each value of model arguments (contained in `trace`) and `proposal_args`, the `involution` function applies an involution that maps the tuple `(get_choices(trace), fwd_choices)` to the tuple `(get_choices(new_trace), bwd_choices)`.
 Note that `fwd_ret` is a deterministic function of `fwd_choices` and `proposal_args`.
 When only discrete random choices are used, the `weight` must be equal to `get_score(new_trace) - get_score(trace)`.
 
-When continuous random choices are used, the `weight` returned by the involution must include an additive correction term that is the determinant of the the Jacobian of the bijection on the continuous random choices that is obtained by currying the involution on the discrete random choices (this correction term is automatically computed if the involution is constructed using the [Involution DSL](@ref)).
+When continuous random choices are used, the `weight` returned by the involution must include an additive correction term that is the determinant of the the Jacobian of the bijection on the continuous random choices that is obtained by currying the involution on the discrete random choices (this correction term is automatically computed if the involution is constructed using the [Bijection DSL](@ref)).
 NOTE: The Jacobian matrix of the bijection on the continuous random choices must be full-rank (i.e. nonzero determinant).
 The `check` keyword argument to the involution can be used to enable or disable any dynamic correctness checks that the involution performs; for successful executions, `check` does not alter the return value.
 """
 function metropolis_hastings(
         trace, proposal::GenerativeFunction,
-        proposal_args::Tuple, involution;
+        proposal_args::Tuple, involution::TraceTransformDSLProgram;
```

Comment:
Does this mean we've gotten rid of any support for involution MH which does not use the trace transform DSL?

This isn't a huge deal (it's pretty easy to implement the old version of involution MH manually if needed), but there may be cases where for performance reasons users would rather manually write an involution (and manually specify the Jacobian calculation) rather than using the DSL.  As a result, I wanted to comment that it might be worth maintaining support for involutions specified as Julia functions.

Reply:
Yes. I realized this would remove support for code based on manual implementation of involutions. Maybe the best way of adding that support is allowing users to construct a SymmetricTraceTranslator using a custom involution not produced using the Trace Transform DSL. The user would be responsible for providing a function that implements something like what is now called `symmetric_trace_translator_run_transform`.

Reply:
@marcoct I think that proposal makes sense.  Another possibility would be to have 1 version of `mh` which accepts a `SymmetricTraceTranslator`, and 1 version which accepts a tuple `(proposal, proposal_args, involution::Function)`.  (This is what I have been using in a fork of this for my open universe modeling.)  This second method could be used with manual involutions.  To retain the current syntax in this PR, we could have another method accepting a `(proposal, proposal_args, involution::TraceTranslatorDSLProgram)` which gets converted to the `SymmetricTraceTranslator` method internally.

Reply:
I also think it makes sense to have a version where manual involutions can be specified! It'd help with both potential performance improvements, and also backward compatibility with older involution MCMC code.

Reply:
I added back for support for involutions defined as Julia functions. It should be backwards-compatible now.